### PR TITLE
Generate config file

### PR DESCRIPTION
**Pull request by jodh-intel**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,40 +1,71 @@
-DESTDIR :=
+DEST_DIR :=
```
Comment (dlespiau) [inline Makefile]: These are confusing. The rest of the world is using DESTDIR, BINDIR, ...

File: Makefile [hunk 2]
```diff
@@ -1,40 +1,71 @@
...
-SYSCONFDIR := $(PREFIX)/etc
-LIBEXECDIR := $(PREFIX)/libexec
+BIN_DIR := $(PREFIX)/bin
+VAR_DIR := $(PREFIX)/var
+SHARE_DIR := $(PREFIX)/share
+SYSCONF_DIR := $(PREFIX)/etc
+LIBEXEC_DIR := $(PREFIX)/libexec
+
+CC_DIR := "clear-containers"
+SHARE_DIR_FULL := $(SHARE_DIR)/$(CC_DIR)
```
Comment (dlespiau) [inline Makefile]: PKGDATADIR?

File: Makefile (hunk 1 above)
Comment (jodh-intel) [inline Makefile]: Ironic as they were trying to be more legible, but ok I'll change these to the common values.

File: Makefile (hunk 2 above)
Comment (jodh-intel) [inline Makefile]: `IGUESSSOFORCONSISTENCYWITHTHESTANDARDSONESBUTTHESESORTOFVARIABLESAREGOINGTOSTARTBECOMINGDIFFICULTTOREAD` ;)

File: Makefile (hunk 1 above)
Comment (dlespiau) [inline Makefile]: Yes, it is a bit ironic, can't really win in either case I guess :/

**Comment by jodh-intel** [comment]
@dlespiau - thanks for reviewing, branch updated.
@devimc - could you take a look?

**Comment by sboeuf** [comment]
LGTM

**Comment by devimc** [comment]
lgtm

**Comment by devimc** [comment]
but semaphore failed

**Comment by jodh-intel** [comment]
Hi @devimc - I think it's failing as the config file sets `BINDIR` to `/usr/local/` which is fine. But that means it will look for `qemu-lite` there too (which is wrong as OBS installs it to `/usr/bin/`). I've just added a `QEMUBINDIR` to allow that behaviour to be changed.

**Comment by jodh-intel** [comment]
Unless Coveralls is counting toml files as source code, it seems confused... ;)

**Comment by sboeuf** [comment]
@jodh-intel still failing, you should have:
```
QEMUBINDIR=/usr/bin
```

**Comment by jodh-intel** [comment]
Yes, the problem is I've got to make that change in https://github.com/clearcontainers/tests to get this PR pass under Semaphore.

**Comment by sboeuf** [comment]
@jodh-intel Ah yes.... https://github.com/clearcontainers/tests introduces more indirection, hence more complexity...

**Comment by jodh-intel** [comment]
https://github.com/clearcontainers/tests/pull/73 raised to hopefully convince the `tests` repo to dtrt here.

**Comment by jodh-intel** [comment]
Blocked on https://github.com/clearcontainers/tests/pull/73.

**Comment by jodh-intel** [comment]
Semaphore failure blocked on https://github.com/clearcontainers/tests/pull/74.

**Comment by sameo** [comment]
LGTM

File: Makefile [hunk 3]
```diff
@@ -1,40 +1,74 @@
...
 BINDIR := $(PREFIX)/bin
+QEMUBINDIR := $(BINDIR)
 SYSCONFDIR := $(PREFIX)/etc
 LIBEXECDIR := $(PREFIX)/libexec
+VARDIR := $(PREFIX)/var
+SHAREDIR := $(PREFIX)/share
+
+CCDIR := clear-containers
+PKGDATADIR := $(SHAREDIR)/$(CCDIR)
+PKGLIBDIR := $(LIBEXECDIR)/$(CCDIR)
```
Comment (dlespiau) [inline Makefile]: PKGLIBEXECDIR

File: Makefile [hunk 4]
```diff
@@ -1,40 +1,74 @@
 DESTDIR :=
 PREFIX := /usr/local
 BINDIR := $(PREFIX)/bin
+QEMUBINDIR := $(BINDIR)
 SYSCONFDIR := $(PREFIX)/etc
 LIBEXECDIR := $(PREFIX)/libexec
+VARDIR := $(PREFIX)/var
```
Comment (dlespiau) [inline Makefile]: LOCALSTATEDIR

**Comment by jodh-intel** [comment]
@dlespiau - ta. Branch updated but still blocked on https://github.com/clearcontainers/tests/pull/74.

**Comment by dlespiau** [comment]
lgtm

**Pull request merged**